autonomy_core/map_plan/mapper/src/test_voxel_mapper.cpp

```diff
@@ -0,0 +1,428 @@
+#include <iostream>
+#include <gtest/gtest.h>
+#include <memory>
+#include <cmath>
+#include <chrono>
+#include <planning_ros_msgs/VoxelMap.h>
+#include "mapper/voxel_mapper.h"
+
+
+// Test class needs to be in the same namespace as VoxelMapper to be able to make it a friend class
+namespace mapper
+{
+
+/*** Declaration of fixture class to use the same configuration across tests ***/
+class VoxelMapperTest : public testing::Test
+{
+protected:
+    // Google test needs to create objects of this test fixture and it calls the default constructor
+    VoxelMapperTest();
+    VoxelMapperTest(double x_orig, double y_orig, double z_orig, int x_dim, int y_dim, int z_dim,
+                    double res, int decay_times);
+
+    const double x_origin_; 
+    const double y_origin_; 
+    const double z_origin_; 
+    const int x_dim_;
+    const int y_dim_;
+    const int z_dim_;
+    const double resolution_;
+    const int decay_times_;
+
+    // This is the VoxelMapper object that is used for all the tests
+    std::unique_ptr<mapper::VoxelMapper> p_test_mapper_;
+
+    // Timing variables to measure method callback performance
+    std::chrono::time_point<std::chrono::steady_clock> time_start, time_end;
```

Comment:
Timing should be done in benchmark not int test.

Reply:
I think I misunderstood this [comment](https://github.com/KumarRobotics/kr_autonomous_flight/issues/35#issuecomment-1138656323). I though you meant to add the timing stuff around the function callback inside the testing code. So did you to just add it while running the test but not include it in the commit?

Reply:
You could either do a microbenchmark by using google benchmark, or you just run the full stack and look at the timing.
Either way, this should not be part of the unit test.

Reply:
ok, I'll remove now then